### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by thaJeztah**
Thought I'd do a round of updating dependencies, and where possible use tagged versions

Also formatting the vendor.conf file, and pin all deps by git-sha, to make it better readable, and to encourage pinning by sha, but align to a tagged release.

- https://github.com/pkg/errors/compare/v0.8.0...v0.8.1
- https://github.com/containerd/console/compare/2748ece16665b45a47f884001d5831ec79703880...0650fd9eeb50bab4fc99dceb9f2e14cf58f36e7f
  - containerd/console#27 console_linux: Fix race: lock Cond before Signal
- https://github.com/docker/go-units/compare/v0.2.0...v0.3.3
  - docker/go-units#8 Enhance FromHumanSize to parse float64 string
  - docker/go-units#20 Add `HumanSizeWithPrecision` function
- https://github.com/urfave/cli/compare/d53eb991652b1d438abdd34ce4bfa3ef1539108e...v1.20.0
  - previous version was somewhere between v1.18 and v1.19
- https://github.com/syndtr/gocapability/compare/db04d3cc01c8b54962a58ec7e491717d06cfcc16...d98352740cb2c55f81556b63d4a1ec64c5a319c2
  - syndtr/gocapability#14 capability: Deprecate NewPid and NewFile for NewPid2 and NewFile2
  - syndtr/gocapability#16 Fix capHeader.pid type
- https://github.com/mrunalp/fileutils/compare/ed869b029674c0e9ce4c0dfa781405c2d9946d08...7d4729fb36185a7c1719923406c9d40e54fb93c7
  - no significant changes, other than some linting fixes
- https://github.com/coreos/go-systemd/compare/v14..v19
  - coreos/go-systemd#248 dbus: add SetPropertiesSubscriber method
  - coreos/go-systemd#251 activation: add support for listeners with names
  - coreos/go-systemd#296 dbus: Fix API break from godbus
- https://github.com/godbus/dbus/compare/v3..v5.0.1
  - godbus/dbus#89 introduce MakeVariantWithSignature
- https://github.com/golang/protobuf/compare/18c9bb3261723cd5401db4d0c9fbc5c3b6c70fe8...v1.0.0
  - picking the first tagged release (newer versions exist, but I have bad experiences with bumping, lol)

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @crosbymichael @cyphar @hqhq rebased; PTAL

**Comment by hqhq** [comment]
CI failed :(

**Comment by thaJeztah** [comment]
hmm...

```
not ok 53 spec validator
# (in test file tests/integration/spec.bats, line 78)
#   `[ "$status" -eq 0 ]' failed
```

Ah, looks like the script doesn't like the new format;

```
  SPEC_COMMIT=$(grep '^github.com/opencontainers/runtime-spec' ${TESTDIR}/../../vendor.conf | cut -d ' ' -f 2)
  run git -C src/runtime-spec reset --hard "${SPEC_COMMIT}"

  [ "$status" -eq 0 ]
```

File: tests/integration/spec.bats [hunk 1]
```diff
@@ -72,7 +72,7 @@ function teardown() {
   run git clone https://github.com/opencontainers/runtime-spec.git src/runtime-spec
   [ "$status" -eq 0 ]
 
-  SPEC_COMMIT=$(grep '^github.com/opencontainers/runtime-spec' ${TESTDIR}/../../vendor.conf | cut -d ' ' -f 2)
+  SPEC_COMMIT=$(grep '^github.com/opencontainers/runtime-spec' ${TESTDIR}/../../vendor.conf | tr -s ' ' | cut -d ' ' -f 2)
```
Comment (thaJeztah) [inline tests/integration/spec.bats]: Think that should fix it 🤞

**Comment by hqhq** [comment]
LGTM

**Comment by cyphar** [comment]
LGTM.

**Comment by cyphar** [comment]
Wait, why do you want to only pin-by-sha? What's wrong with using versions?

**Comment by thaJeztah** [comment]
> What's wrong with using versions?

Nothing really wrong with version, but this makes the format more consistent, and makes sure things don't change if a dependency decides to re-tag a version (not an imaginary scenario; we ran into this)

**Comment by justincormack** [comment]
Needs a rebase

**Comment by thaJeztah** [comment]
Rebased; PTAL

**Comment by AkihiroSuda** [comment]
ping @cyphar

**Comment by AkihiroSuda** [comment]
> What's wrong with using versions?


Versions might be modified when a maintainer's account got hacked, while sha won't be modified

**Comment by AkihiroSuda** [comment]
needs rebase

**Comment by thaJeztah** [comment]
rebased again (one dependency was dropped in https://github.com/opencontainers/runc/pull/2116)

**Comment by mrunalp** [comment]
Needs rebase again

**Comment by thaJeztah** [comment]
rebased

**Comment by mrunalp** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

**Pull request merged**